aa_forum/models.py

```diff
@@ -205,57 +180,59 @@ class Boards(models.Model):
         related_name="aa_forum_boards_group_restriction",
     )
     order = models.IntegerField(default=0)
+    first_message = models.ForeignKey(
+        "Message",
+        editable=False,
+        null=True,
+        default=None,
+        related_name="+",
+        on_delete=models.SET_DEFAULT,
+        help_text="Shortcut for better performance",
+    )
+    last_message = models.ForeignKey(
+        "Message",
+        editable=False,
+        null=True,
+        default=None,
+        related_name="+",
+        on_delete=models.SET_DEFAULT,
+        help_text="Shortcut for better performance",
+    )
 
     class Meta:
         """
         Meta definitions
         """
 
         default_permissions = ()
-        verbose_name = _("Board")
-        verbose_name_plural = _("Boards")
-
-    def save(
-        self, force_insert=False, force_update=False, using=None, update_fields=None
-    ):
-        """
-        Add the slug on save
-        """
-
-        if self.slug is None or self.slug == "":
-            slug = get_slug_on_save(subject=self.name)
-            self.slug = slug
-
-        super().save()
+        verbose_name = _("board")
+        verbose_name_plural = _("boards")
+        constraints = [
+            models.UniqueConstraint(fields=["category", "name"], name="fpk_board")
+        ]
 
     def __str__(self) -> str:
         return str(self.name)
 
-    def latest_topic(self):
-        """
-        Get the latest topic for this board
-        :return:
-        :rtype:
-        """
-
-        topic = Topics.objects.filter(board=self).order_by("time_modified").last()
+    @transaction.atomic()
+    def save(self, *args, **kwargs):
+        # Add the slug on save if it does not exist
+        try:
+            self.slug
+        except ObjectDoesNotExist:
+            self.slug = get_slug_on_save(subject=self.name)
+        super().save(*args, **kwargs)
```

Comment:
Saving a board with group restrictions doesn't work. Groups are not applied. Neither when editing a board.

Reply:
Gefixed mit 61b2304